hw/ip_templates/gpio/dv/env/gpio_scoreboard.sv.tpl

```diff
@@ -384,33 +384,29 @@ class ${module_instance_name}_scoreboard extends cip_base_scoreboard #(.CFG_T ($
   virtual task monitor_gpio_straps();
     logic [NUM_GPIOS-1:0] gpio_i_sampled;
     forever begin : monitor_gpio_straps
-      // Wait for going out of reset operation.
-      wait(!cfg.under_reset);
-      // Wait until the strap_en input be triggered
-      // if a reset comes in the middle, step-out of the loop.
-      while (!cfg.straps_vif_inst.tb_port.strap_en) begin
-        cfg.clk_rst_vif.wait_clks_or_rst(1);
-        if (cfg.under_reset) break;
+      @(cfg.straps_vif_inst.tb_port.strap_en or cfg.under_reset) begin
+        wait(!cfg.under_reset);
+        // Get the gpio_i input data from the pins interface.
+        gpio_i_sampled = cfg.gpio_vif.pins;
+        // Wait for one clock cycle to update the register model.
+        cfg.clk_rst_vif.wait_clks(1);
```

Comment:
Oof, this isn't great: it will only make sense if we don't get a reset asserted in the middle of a clock cycle. I think a cleaner approach would probably be something like this:
```systemverilog
fork begin : isolation_fork
  fork
    wait(cfg.under_reset);
    forever
      @(cfg.straps_vif_inst.tb_port.strap_en);
      gpio_i_sampled = ...;
      cfg.clk_rst_vif.wait_clks(1);
      /// some more stuff
    end
  join_any
  disable fork;
end join
```